lnd.go

```diff
@@ -351,23 +376,115 @@ func fileExists(name string) bool {
 	return true
 }
 
-// genCertPair generates a key/cert pair to the paths provided.
-// This function is adapted from https://github.com/btcsuite/btcd
+// genCertPair generates a key/cert pair to the paths provided. The
+// auto-generated certificates should *not* be used in production for public
+// access as they're self-signed and don't necessarily contain all of the
+// desired hostnames for the service. For production/public use, consider a
+// real PKI.
+//
+// This function is adapted from https://github.com/btcsuite/btcd and
+// https://github.com/btcsuite/btcutil
 func genCertPair(certFile, keyFile string) error {
 	rpcsLog.Infof("Generating TLS certificates...")
 
 	org := "lnd autogenerated cert"
-	validUntil := time.Now().Add(autogenCertValidity)
-	cert, key, err := btcutil.NewTLSCertPair(org, validUntil, nil)
+	now := time.Now()
+	validUntil := now.Add(autogenCertValidity)
+
+	priv, err := rsa.GenerateKey(rand.Reader, 4096)
 	if err != nil {
 		return err
 	}
 
+	// end of ASN.1 time
+	endOfTime := time.Date(2049, 12, 31, 23, 59, 59, 0, time.UTC)
```

Comment:
These are initialized with function calls, so they have to be `var`s, not `const`s but I broke them out into the `var` section at the top of the file.